### PR TITLE
Add tests to the core gpp code

### PR DESCRIPTION
**Pull request by hhhjort**

### CODE REVIEW

File: util/bitstream.go [hunk 1]
```diff
@@ -116,7 +116,7 @@ func ParseByte6(data []byte, bitStartIndex uint16) (byte, error) {
 		return data[startByte] >> (2 - bitStartIndex), nil
 	}
 	if uint16(len(data)) < (startByte + 2) {
-		return 0, fmt.Errorf("Expected 6 bits to start at bit %d, but the byte array was only %d bytes long", bitStartIndex, len(data))
+		return 0, fmt.Errorf("Expected 6 bits to start at bit %d, but the byte array was only %d bytes long (needs second byte)", bitStartIndex, len(data))
```
Comment (SyntaxNode) [inline util/bitstream.go]: Let's try to follow proper Go naming conventions this time around. In which case, error strings should not start with a capitalized letter.

File: util/bitstream_test.go [hunk 2]
```diff
@@ -16,6 +16,28 @@ type testDefinition struct {
...
+	{testdata, 5, 1},  // testdata 1 in first byte
+	{testdata, 23, 1}, // testdata 1 in last bit of third byte
+	{testdata, 47, 1}, // testdata 1 in last bit of last byte
+}
+
+func TestReadByte1(t *testing.T) {
+	bs := BitStream{b: testdata, p: 80}
+
+	b, err := bs.ReadByte1()
+	assertStringsEqual(t, "Expected 1 bit at bit 80, but the byte array was only 6 bytes long", err.Error())
```
Comment (SyntaxNode) [inline util/bitstream_test.go]: Nitpick: Do we need the assert helper methods? Can we use `assert.Error(t, err, "string")` instead?

File: util/bitstream_test.go [hunk 3]
```diff
@@ -16,6 +16,28 @@ type testDefinition struct {
...
+
+	for _, test := range test1Bit {
+		bs = BitStream{b: test.data, p: test.offset}
```
Comment (SyntaxNode) [inline util/bitstream_test.go]: I discovered a testing pattern we haven't used before. We might be better off using t.Run(description, anon func) to isolate each test in the loop. There's some cool tooling around the use of t.Run as well which might be useful to us in the future.

File: util/bitstream_test.go [hunk 4]
```diff
@@ -27,15 +49,47 @@ var test4Bits = []testDefinition{
...
+	{[]byte{0x92, 0x80}, 4, 10}, // Offset which aligns with a nibble
+	{[]byte{0x99}, 1, 76},       // Offset which doesn't align with a nibble.
+	{[]byte{0x01, 0xe0}, 7, 60}, // Offset which involves flowing over to a second byte
+}
+
+func TestReadByte6(t *testing.T) {
+	bs := BitStream{b: testdata, p: 46}
+
+	b, err := bs.ReadByte6()
+	assertStringsEqual(t, "Expected 6 bits to start at bit 46, but the byte array was only 6 bytes long (needs second byte)", err.Error())
```
Comment (SyntaxNode) [inline util/bitstream_test.go]: Why are these cases not part of the table?

File: util/bitstream_test.go (hunk 2 above)
Comment (hhhjort) [reply util/bitstream_test.go]: I am already in the process of removing all the assert helper methods. :)

File: util/bitstream_test.go (hunk 3 above)
Comment (hhhjort) [reply util/bitstream_test.go]: That is also in process. :)

File: util/bitstream_test.go (hunk 4 above)
Comment (hhhjort) [reply util/bitstream_test.go]: Need a separate testing function for the errors. I had been pondering tabling them as well though.

Comment (SyntaxNode) [reply util/bitstream_test.go]: Might be a good idea.

File: util/fibonacci_test.go [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+}
+
+func TestReadFibonnaciInt(t *testing.T) {
+	bs := BitStream{b: testData, p: 47}
+	i, err := bs.ReadFibonacciInt()
+	assert.Equal(t, "error reading bit 2 of Integer(Fibonacci): expected 1 bit at bit 48, but the byte array was only 6 bytes long", err.Error())
+
+	// bs = BitStream{b: testData, p: 40}
+	// i, err = bs.ReadUInt16()
+	// assertStringsEqual(t, "Expected a 16-bit int to start at bit 40, but the byte array was only 6 bytes long", err.Error())
```
Comment (guscarreon) [inline util/fibonacci_test.go]: We probably don't need these three lines anymore

**Review by guscarreon — APPROVED** [review]
LGTM

**Pull request merged**